scanpy/external/tl/_sam.py

```diff
@@ -0,0 +1,216 @@
+"""\
+Run the Self-Assembling Manifold algorithm
+"""
+from typing import Optional, Union
+
+from anndata import AnnData
+
+from ... import logging as logg
+
+
+def sam(
+    adata: AnnData,
+    max_iter: int = 10,
+    num_norm_avg: int = 50,
+    k: int = 20,
+    distance: str = 'correlation',
+    standardization: Optional[str] = 'Normalizer',
```

Comment:
```suggestion
    standardization: Literal['Normalizer', 'StandardScaler'] = 'Normalizer',
```

Please do `from ..._compat import Literal` or so to get this.